c++/src/ColumnPrinter.cc

```diff
@@ -252,7 +256,7 @@ namespace orc {
       break;
 
     case STRUCT:
-      result = new StructColumnPrinter(buffer, type);
+      result = new StructColumnPrinter(buffer, type, selectedColumns);
```

Comment:
Actually, this is the intended implementation: only specify top-level (logical) columns. Otherwise, there is no way to distinguish between logical and physical columns.